### PR TITLE
Add Flush Interval to default Buffered Logger 

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds a default flush interval of 15 seconds to the BufferedLogger type to avoid logs being held until the job ends. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Review by shunping — APPROVED** [review]

File: sdks/python/container/boot.go [hunk 1]
```diff
@@ -278,7 +280,7 @@ func launchSDKProcess() error {
 		go func(workerId string) {
 			defer wg.Done()
 
-			bufLogger := tools.NewBufferedLogger(logger)
+			bufLogger := tools.NewBufferedLoggerWithFlushInterval(context.Background(), logger, loggerFlushInterval)
```
[the remark this answers is not in the document]
Comment (jrmccluskey) [reply sdks/python/container/boot.go:283]: Having looked at it we do use both instantiation methods in different places. We could clean things up and unify those paths if we really, really felt the need to but in this context that seems unnecessary. This kind of overloaded constructor pattern isn't atypical in Go, and I think the distinct methods provide a pretty clear indication of what you're getting when you use them (buffering vs buffering with a regular flush interval.) 

If there was another code change to make it'd probably be to add some sort of flushing behavior every X logging entries stored, but that may be overcomplicating things in the case of dumping long tracebacks for errors.

[the remark this answers is not in the document]
Comment (tvalentyn) [reply sdks/python/container/boot.go:283]: My point is less about code style but that we shouldn't be using buffered logger without flushing ever, looks like this is the last place where we do so: https://github.com/search?q=repo%3Aapache%2Fbeam%20NewBufferedLogger&type=code

[the remark this answers is not in the document]
Comment (jrmccluskey) [reply sdks/python/container/boot.go:283]: Looking at the code more I'm a little surprised that there isn't any sort of flush mechanism beyond the time. I think we wound up dropping the buffered logging into the worker process startup to catch any early errors and it wound up swallowing too much. 10-15 seconds as a default should be fine then

I may try to come back to the buffered logging at some point when I have more time and try to do some other polishing, I have a few ideas as to why outputs from python segfaults get all screwy now.

**Comment by tvalentyn** [comment]
LGTM. Please test this though.

Optionally, consider using NewBufferedLogger where the other constructor is used, if it makes sense or keep as is.

**Pull request merged**